docs/source/development/style-guide/naming-conventions.rst

```diff
@@ -3,6 +3,55 @@
 Naming Conventions
 ------------------
 
+Data Product File Naming Conventions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following describes the filename conventions for IMAP data products. These conventions follow the `recommendations
+provided by the SPDF <https://spdf.gsfc.nasa.gov/guidelines/filenaming_recommendations.html>`_.
+
+The general filename convention is as follows::
+
+    imap_<instrument>_<datalevel>_<descriptor>_<startdate>_<enddate>_<version>.<format>
+
+* ``<instrument>`` is the IMAP instrument associated with the data product. Acceptable values include: ``codice``,
+  ``glows``, ``hi-45``, ``hi-90``, ``hit``, ``idex``, ``lo``, ``mag``, ``swapi``, ``swe``, ``ultra-45``, and
+  ``ultra-90``
+
+* ``<datalevel>`` is the data level for the data product.  Acceptable values depend on the instrument:
+
+  * ``codice``: ``l0``, ``l1a``, ``l1b``, ``l2``, ``l3``
+  * ``glows``: TBD
+  * ``hi-45``: TBD
+  * ``hi-90``: TBD
+  * ``hit``: TBD
+  * ``idex``: TBD
+  * ``lo``: TBD
+  * ``mag``: TBD
+  * ``swapi``: TBD
+  * ``swe``: TBD
+  * ``ultra-45``: TBD
+  * ``ultra-90``: TBD
```

Comment:
Should we explicitly list the supported data levels here?  If so, I will need to dig through our documentation add these. (If anyone knows what they are for the instrument(s) they are working on and could comment here, that would be helpful!)

Reply:
MAG: L1A, L1B, L1C, L2pre, L2
GLOWS: L1A, L1B, L2, L3a, L3b, L3c, L3d

That being said, I don't think they need to be included, because I don't think we're checking that in the code...?